### ISSUE TITLE
Can't specify remote as argument when using `push`

### ISSUE THREAD

**Issue by loris** [jamiew/git-friendly#54]
It looks like `push` is only pushing to the remote set in the `git config` and does not allow to specify the remote at run time like `git push [remote] [branch]` allow.
It is quite useful when you are using hosting like Beanstalk or Heroku where you can deploy your app by git pushing on their remote too.

**Comment by sapegin** [comment]
You’re right, and I’m afraid it maybe tricky to implement because of this feature:

> Any extra args to this command will be passed through to 'git push',
> e.g. for doing "push -f"

But feel free to send a PR ;-)

**Comment by jm3** [comment]
owch, I also hit this one today.

@sapegin maybe I'm missing something, but if i have two named remotes that work fine when pushed to with bare git, e.g. `git push dropbox` or `git push github`, then, if subsequent args get passed through to git push, why doesn't wouldn't a plain`push github` work?

I get the error:

```
➜ ~/Code/hak (master) > git remote
dropbox
github
➜ ~/Code/haz (master) > push github
Ouch, push failed!

fatal: refs/remotes/dropbox/HEAD cannot be resolved to branch
```

**Comment by jamiew** [comment]
HMMMM nice find Mr. Manoogian

Testing this out locally too

**Comment by jamiew** [comment]
OK so I've figured out why this happening.

We're calling `git push` like so:

```
git push --set-upstream $* $remote $remote_branch
```

and `$remote` and `$remote_branch` are being "intelligently" populated, so you end up with something like

```
~/dev/blackbook(master)$ push github
git push --set-upstream github github master
```

If there's an upstream set, the `$remote $remote_branch` is not actually required. Maybe we could do something clever like set the default upstream if missing, and then just do `git push $*`?

**Comment by jm3** [comment]
Crude testing: commits push to alternate remote as advertised! Thanks, @jamiew!

[![asciicast](https://asciinema.org/a/FhL9wDvBikPJ0TKMng7HRiUZO.svg)](https://asciinema.org/a/FhL9wDvBikPJ0TKMng7HRiUZO)

**Comment by jamiew** [comment]
Awesome, glad it's working for you too. I'm gonna keep testing for a few more days before fully merging. I'm also wondering if this might fix #65!

**Comment by jamiew** [comment]
OK been using this for a few weeks and think we are gtg, gonna open a proper pull request and merge!

I'd cut a release too if we were cutting releases! Something something homebrew package I keep meaning to add!
